sdk/resourcemanager/azure-resourcemanager-network/src/test/java/com/azure/resourcemanager/network/ApplicationGatewayTests.java

```diff
@@ -682,6 +682,46 @@ public void canSetSslPolicy() {
         });
     }
 
+    @Test
+    public void canCreateApplicationGatewayWithDefaultSku() {
+        String appGatewayName = generateRandomResourceName("agw", 15);
+
+        String appPublicIp = generateRandomResourceName("pip", 15);
+
+        PublicIpAddress pip =
+            networkManager
+                .publicIpAddresses()
+                .define(appPublicIp)
+                .withRegion(Region.US_EAST)
+                .withNewResourceGroup(rgName)
+                .withSku(PublicIPSkuType.STANDARD)
```

Comment:
Does AppGateway work with BASIC ipAddress?

Reply:
No it doesn't... It's more like a V2 SKU to me.
> Application Gateway with SKU Basic can only reference public ip with Standard SKU

Reply:
Got it. Do you know whether `withNewPublicIpAddress()` creates an IP address?

Reply:
Yeah, it'll be created in `createInner`, through `ensureDefaultPipDefinition`:
https://github.com/Azure/azure-sdk-for-java/blob/654650746b7294af2f81457cde7db9afbbee76e2/sdk/resourcemanager/azure-resourcemanager-network/src/main/java/com/azure/resourcemanager/network/implementation/ApplicationGatewayImpl.java#L563-L573

Maybe we'll set pip sku to `STANDARD` here if it's not legacy gateway.

Reply:
Maybe next year. As currently user can still create V1 (if subscription has V1)